cachito/workers/pkg_managers/general_js.py

```diff
@@ -620,8 +605,37 @@ def upload_non_registry_dependency(
         nexus.upload_asset_only_component(repo_name, "npm", modified_dep_archive)
 
 
-def _fetch_external_dep(dep_identifier: str, temp_dir: str) -> str:
-    """Fetch an external (git or https) JS dependency."""
+def _fetch_external_dep(dep_identifier: str, is_git: bool, temp_dir: str) -> str:
+    """Fetch an external (git or https) JS dependency.
+
+    If the dependency is a git dependency, raise an exception if dangerous scripts are present in
+    the package.json file.
+
+    A "dangerous" script is one that needs to be executed in order to ensure that the package
+    content is correct. We can avoid executing them, but by not executing them we would run the risk
+    of serving incorrect content to the users.
+    """
+    log.info("Downloading the npm dependency %s to be uploaded to Nexus", dep_identifier)
+
+    if is_git:
+        archive_path = _clone_git_dep(dep_identifier)
+        _check_dangerous_scripts(dep_identifier, archive_path)
+
+        shutil.unpack_archive(archive_path, temp_dir)
+        repo_path = os.path.join(temp_dir, "app")
+        # Use npm pack on the unpacked repo directory to apply npm's file exclusion logic.
+        # By packing a directory (rather than an archive), we ensure npm will use the DirFetcher,
+        # which uses npm-packlist to figure out what belongs in the final packed tarball.
+        # https://github.com/npm/pacote/blob/18e760f5c438155c1b8af8aa1ffbead874732058/lib/dir.js#L82
+        # https://github.com/npm/npm-packlist/blob/main/lib/index.js
+        # In fact, GitFetcher simply clones the repo, does a bit of preparation (npm install) and
+        # then delegates to DirFetcher. By using DirFetcher directly, we avoid the npm install.
+        # The purpose of npm install is to install the dependencies that may be needed for the
+        # "dangerous" scripts, which we're not going to execute anyway.
+        npm_pack_args = ["npm", "pack", f"file:{repo_path}"]
```

Comment:
Does this avoid executing scripts in transitive dependencies for the specific git dependency we're handling here?

Reply:
It does. When packing a directory, npm doesn't call `npm install`, so the transitive dependencies don't even get downloaded

Reply:
(and I think transitive dependencies were never really a problem, those respect NPM_CONFIG_IGNORE_SCRIPTS)